src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -34,14 +35,13 @@ void AddSuppression (CustomAttribute ca, ICustomAttributeProvider provider)
 		void AddSuppression (SuppressMessageInfo info, ICustomAttributeProvider provider)
 		{
 			if (!_suppressions.TryGetValue (provider, out var suppressions)) {
-				suppressions = new Dictionary<int, SuppressMessageInfo> ();
+				suppressions = (ScannedProvider: false, Suppressions: new Dictionary<int, SuppressMessageInfo> ());
```

Comment:
Did you consider just always scanning the provider when adding a record to the cache?
It is potentially "slightly" wasteful, but it would reduce the complexity of this code.
I personally would prefer that to the "half populated" cache solution.

Reply:
I took the approach we discussed where we populate the cache for all members referenced by the assembly-level attributes. PTAL.